python/mlx/optimizers/schedulers.py

```diff
@@ -84,3 +85,33 @@ def scheduler(step):
         return init * decay
 
     return scheduler
+
+
+def linear_warmup(
+    schedule_fn, length: int, finish: float, init: float = 0.00
+) -> Callable:
+    """
+    >>> lr_schedule = optim.cosine_with_warmup(100, 1e-1)
+    >>> optimizer = optim.Adam(learning_rate=lr_schedule)
+    >>> optimizer.learning_rate
+    array(0.0, dtype=float32)
+    >>> for _ in range(100): optimizer.update({}, {})
+    ...
+    >>> optimizer.learning_rate
+    array(0.1, dtype=float32)
+
+    :param schedule_fn: other schedule function, such as cosine_decay,step_decay, etc.
+    :param length: warmup length
+    :param init: start value (0 by default)
+    :param finish: final value
+    :return: callable takes a step and returns the schedulable
```

Comment:
Could you follow the same docstring format as other functions in this package?

Reply:
Yes, and I'll also try for a clean implementation of the schedule joining you mentioned.  Seems very helpful to have.

Reply:
Awesome, thanks!